lib/gstreamer/transcoderbase.py

```diff
@@ -5,9 +5,11 @@
 ###
 
 import slash
+import os
 
 from ...lib.common import timefn, get_media, call, exe2os, filepath2os
 from ...lib.gstreamer.util import have_gst, have_gst_element, gst_discover
+from ...lib.gstreamer.msdk.util import mapformatu
```

Comment:
It looks like we have a common format mapping functions (mapformat, mapformatu, map_best_hw_format) in gst-msdk, gst-va and gst-vaapi util separately, how about moving these format-mapping functions into lib.gstreamer.util?

Reply:
Does it hurt to add "BGRX" to the format mapping list? We actually don't set BGRx in gst-va pipeline, right?

Reply:
The test case definitions are generic.  Look in FULL vpp csc cases.  The gst-va/vaapi tests will automatically skip the BGRx cases because it is not in it's supported format lookup table.  If we add it to their lookup tables, then they won't be skipped... you will need to verify they don't fail in that case.

Reply:
...it's the reason why each one is defined separately.  That is, not all of the plugins support the same set of formats.

Reply:
Ok, then I added BaseFormatMapper class in lib.gstreamer.util.py same like ffmpeg did. 
One thing I am not sure if we need to add p010-10le mapping result for P010 in this lookup table, because the table in gstreamer BaseFromatMapper is currently only used by transcoderbase and in transcoding case we don't use rawvideoparse which needs lowercase letters...

Reply:
Another question about supported format in BaseFormatMapper table, I don't include BGRX considering it is a common table that should fits all gst plugins. Any comments on this?

Reply:
> without `mapformatu` load please remove it

I forgot it... Thanks, removed it now.

Reply:
> Ok, then I added BaseFormatMapper class in lib.gstreamer.util.py same like ffmpeg did. One thing I am not sure if we need to add p010-10le mapping result for P010 in this lookup table, because the table in gstreamer BaseFromatMapper is currently only used by transcoderbase and in transcoding case we don't use rawvideoparse which needs lowercase letters...

Later we may use gstreamer BaseFormatMapper in other parts (encode, decode, etc).  But for now, it should be ok without lowercase mapping.  We can refactor later when we need it.  The transcoder test code is getting pretty ugly, old and hard to change... I really want to redesign it but just haven't had time :(

Reply:
> > Another question about supported format in BaseFormatMapper table, I don't include BGRX considering it is a common table that should fits all gst plugins. Any comments on this?
> 
> Maybe this will help to your question
> 
> ```
> class A(object):
>   def test(self):
>     print('aaa')
> 
> class B(object):
>   def test(self):
>     print('bbb')
> 
> class C(A):
>   def test(self):
>     print('ccc')
> 
> class D(C,B):
>   def __init__(self):
>     pass
>   def test(self):  
>     B.test(self)
> 
> obj = D()
> obj.test()  # output is bbb
> ```

Right, the derived class (in msdk/transcoder) can just override `get_supported_format_map` to add more formats...
```python
def get_supported_format_map()
  fmts = super().get_supported_format_map()
  fmts.update(BGRX = "BGRx")
  return fmts
```
...but it's not necessary at this moment for current transcode cases.  Do it later.

Reply:
Thanks for Bin's suggestions! I also agree it is not necessary to do it in this PR. If possible, I can do it in Q4 :)